cmd/abigen/main.go

```diff
@@ -234,11 +238,10 @@ func abigen(c *cli.Context) error {
 			abis = append(abis, string(abi))
 			bins = append(bins, contract.Code)
 			sigs = append(sigs, contract.Hashes)
-			nameParts := strings.Split(name, ":")
-			types = append(types, nameParts[len(nameParts)-1])
+			types = append(types, typeName)
```

Comment:
And I just looked at the diff again. Was it even possible before to have two contracts with the same name? The `types` slice would have duplicate strings, is this supported by `bind.Bind` (called later with this slice)?

Reply:
Would also be interested on your feedback here @karalabe . It was argued before that the full path allows ignoring specific contracts if the same contract name is reused in the same project. This logic here, also present in the old version, makes me wonder if it was even at all supported before to reuse the same contract name.